docs/assignment1.md

```diff
@@ -1,62 +1,80 @@
 
 | ID  | Short name  | Description  |
 |---|---|---|
-| F1  | Tags | Code snippets can be tagged via freely-defined labels called tags  |
-| F2  | Commands  | The player can control the main character by issuing command-line commands following this syntax: `command-name [target-objects]*`. The available `command-names` are the following: <br/> - move: ... <br/> - use: ... <br/> - inspect: ... <br/> |
-| F3  | Movements  | The main character shall move freely in the environment according  |
-| F4  | ... | ... |
+| F1  | Commands  | The player can control the main character by typing command-line instructions following this syntax: `command + [target]`, where the (\*) below indicates that a target is optional. <br/> Available `commands` are: <br/><br/> - `move` + [north, east, south, west, area-name] <br/> - `use` + [item] <br/> - `look` + [item]\* <br/> - `interact` + [npc, item] <br/> - `inventory` <br/> - `save` <br/> - `load` <br/> - `quit` <br/> |
+| F2  | Movements  | The player may move freely between different areas in four directions as described by the `move` command above, or by specifying a unique area name (optionally given in the description of the current area). Some areas may be blocked by obstacles and therefore inaccessible. Obstacles may be removed by using a specific item in their presence (e.g. "use key" might open a door).  |
+| F3  | Areas | All areas will be represented as nodes in a graph, where edges between nodes indicate two areas are reachable from one another. Each area will have a description so the player can distinguish between different type of areas, however, some areas may not be unique.  |
+| F4  | Obstacles | An obstacle is a special type of object which prohibits movement between two specific areas until a certain item is used in the same area that they inhabit. Having obstacles be generic like this allows some more complex behaviour to exhibit itself - for example, defeating a monster may have them drop an item that "unlocks" an obstacle, therefore inhibiting progress until the monster is defeated. |
+| F5  | Items | The player may acquire and use items throughout the game. Items will be stored in an inventory of infinite size, which may be accessed using the `inventory` command (i.e. `inventory` will print the names of the currently held items). Some items are consumed upon first use, whereas other items persist throughout the game. Several commands may be used in combination with an item: </br> - `interact` acquires the item and places it in the player's inventory. </br> - `look` briefly describes the item. </br> - `use` queries the item, asking it what it may be used for, and if appropriate for the situation, uses it (e.g. `use key` might consume the key if in the presence of the appropriate door, otherwise the inventory will remain unchanged).  |
```

Comment:
Technically speaking, infinite inventory size is impossible. It is better to bound it to a maximum value